### PR TITLE
Python: Modernise injection libraries.

### PR DESCRIPTION
**Pull request by tausbn**
Also adds `StringValue` as a new part of the `Value` API. The alternative would be to use `Value::forString` as seen elsewhere, but this is somewhat arcane and unintuitive to use. 

At a later point, we can go through and change all of the relevant instances of that idiom to use `StringValue` instead.

### CODE REVIEW

File: python/ql/src/semmle/python/objects/ObjectAPI.qll [hunk 1]
```diff
@@ -684,6 +705,15 @@ module ClassValue {
         result = TBuiltinClassObject(Builtin::special("unicode"))
     }
 
+    /** Get the `ClassValue` for the `str` class. This is `bytes` in Python 2,
+    and `unicode` in Python 3. */
```
Comment (RasmusWL) [inline python/ql/src/semmle/python/objects/ObjectAPI.qll]: Is this what you meant? :blush: 

```
/** Get the `ClassValue` for the `str` class. This is `ClassValue::bytes` for Python 2,
    and `ClassValue::unicode` for Python 3. */
```

With the current comment, I read it as returning the result of the _python code_ `bytes` for Python 2, and `unicode` for Python 3 (which makes no sense)

File: python/ql/test/library-tests/web/django/Sinks.expected [hunk 2]
```diff
@@ -1,6 +1,8 @@
 | sql.py:13 | Str | externally controlled string |
 | sql.py:14 | Str | externally controlled string |
 | sql.py:17 | BinaryExpr | externally controlled string |
+| sql.py:17 | Str | externally controlled string |
```
Comment (RasmusWL) [inline python/ql/test/library-tests/web/django/Sinks.expected]: Line 17 `cursor.execute("SELECT * FROM users WHERE username = '%s'" % username)`

so does this mean that now we treat the string `"SELECT * FROM users WHERE username = '%s'"` as a _externally controlled string_? and isn't that a bit strange? :confused:

File: python/ql/src/semmle/python/objects/ObjectAPI.qll (hunk 1 above)
Comment (tausbn) [reply python/ql/src/semmle/python/objects/ObjectAPI.qll]: Ah, indeed `unicode` does not make sense. It is intended to return the class corresponding to what you get when you write `str` in Python. In Python 2, this is indeed the same as `bytes` (that is, `str is bytes` is true), and in Python 3 it's... `str`.

I must admit, I just assumed the corresponding comment for `ClassObject theStrType()` in `ClassObject.qll` was correct. (It suffers from the same error.)

File: python/ql/test/library-tests/web/django/Sinks.expected (hunk 2 above)
Comment (tausbn) [reply python/ql/test/library-tests/web/django/Sinks.expected]: Hmm... That _does_ sound a bit fishy. I'll investigate.

Comment (tausbn) [reply python/ql/test/library-tests/web/django/Sinks.expected]: Ah, what this is _actually_ saying is that the node in question is a _sink_ of that particular kind of taint. That's slightly less worrisome, although that now makes me wonder whether `username` should also count as a sink, separately (probably not).

**Comment by tausbn** [comment]
Closing this for now, as it's unlikely to be revisited in the near future.

**Pull request closed without merging**